### ISSUE TITLE
Why OpenSearchStreamingLambdaFunction does not have app name prefix and current env suffix?

### ISSUE THREAD

**Issue by aleksey-shmatov** [aws-amplify/amplify-cli#9480]
**Which Category is your question related to?**
API

**Amplify CLI Version** 
7.6.7

**What AWS Services are you utilizing?**
Amplify

**Provide additional details e.g. code snippets**
Pushing minor updates in schema caused changing search lambda name to `OpenSearchStreamingLambdaFunction` - without app name or environment. Is this expected behaviour?
 Trying to push to production now causes error: 

> OpenSearchStreamingLambdaFunction already exists in stack

**Comment by ngnathan** [comment]
I've got the same issue.  

As of 7.6.7, it changes the previous OpenSearchStreamingLambda function from `amplify-<appName>-<env>-OpenSearchStreamingLambd-<someId>` to `OpenSearchStreamingLambdaFunction`, without any other identifier.

I updated my dev env, but I can no longer push to prod, due to the error:
Following resources failed
Resource Name: amplify-<appName>-<env>-OpenSearchStreamingLambd-<someId> (AWS::Lambda::Function)
Event Type: update
Reason: OpenSearchStreamingLambdaFunction already exists in stack <LambdaArn>

**Comment by mattfysh** [comment]
If possible, can someone from Amplify team recommend a workaround, e.g. do we downgrade the CLI version?

Customers unable to deploy to production environments should be treated as a high-severity issue

**Comment by cjihrig** [comment]
@mattfysh downgrading to a version prior to v7.6.7 should let you workaround the issue.

**Comment by ngnathan** [comment]
> If possible, can someone from Amplify team recommend a workaround, e.g. do we downgrade the CLI version?
> 
> Customers unable to deploy to production environments should be treated as a high-severity issue

@mattfysh I can confirm.  I downgraded to 7.6.5 (including in the Amplify console if you're using Amplify's git-based hosting) and it reverts back to the previous lambda name.  Test first in a non-production env to ensure that the OpenSearchStreamingLambdaFunction reverts back to the previous name.

**Comment by Rafcin** [comment]
Yea this is def a problem, my dev env had the prefixed es lambda, but my main is now `OpenSearchStreamingLambdaFunction` and I can't push any changes

**Comment by Rafcin** [comment]
When I push I get the following errors:
```
UPDATE_FAILED               OpenSearchStreamingLambdaFunction6ED4A1FD                               AWS::Lambda::Function      Wed Jan 12 2022 15:08:34 GMT-0800 (Pacific Standard Time) OpenSearchStreamingLambdaFunction already exists in stack arn:aws:cloudformation:us-west-2:707129987630:stack/amplify-fox-prod-174207-apifox-1XCP8S1CGPDTM-SearchableStack-16OIOH0271T9U/63e749e0-263f-11ec-baa0-0a7614ed120d
UPDATE_FAILED               OpenSearchAccessIAMRole6A1D9CC5                                         AWS::IAM::Role             Wed Jan 12 2022 15:08:35 GMT-0800 (Pacific Standard Time) Resource creation cancelled                                                                                                                                                                                                  
UPDATE_ROLLBACK_IN_PROGRESS amplify-fox-dev-160955-apifox-Y0QXBP993EX1-SearchableStack-TV78E7975PIC AWS::CloudFormation::Stack Wed Jan 12 2022 15:08:36 GMT-0800 (Pacific Standard Time) The following resource(s) failed to update: [OpenSearchAccessIAMRole6A1D9CC5, OpenSearchStreamingLambdaFunction6ED4A1FD].                                                                                                    
⠸ Updating resources in the cloud. This may take a few minutes...

UPDATE_FAILED SearchableStack AWS::CloudFormation::Stack Wed Jan 12 2022 15:08:40 GMT-0800 (Pacific Standard Time) Embedded stack arn:aws:cloudformation:us-west-2:707129987630:stack/amplify-fox-dev-160955-apifox-Y0QXBP993EX1-SearchableStack-TV78E7975PIC/ec47cd50-263a-11ec-bcd6-0a9127743e0d was not successfully updated. Currently in UPDATE_ROLLBACK_IN_PROGRESS with reason: The following resource(s) failed to update: [OpenSearchAccessIAMRole6A1D9CC5, OpenSearchStreamingLambdaFunction6ED4A1FD]. 
⠏ Updating resources in the cloud. This may take a few minutes...

UPDATE_ROLLBACK_IN_PROGRESS amplify-fox-dev-160955-apifox-Y0QXBP993EX1 AWS::CloudFormation::Stack Wed Jan 12 2022 15:08:42 GMT-0800 (Pacific Standard Time) The following resource(s) failed to update: [SearchableStack]. 
⠙ Updating resources in the cloud. This may take a few minutes...

UPDATE_FAILED               apifox                 AWS::CloudFormation::Stack Wed Jan 12 2022 15:08:46 GMT-0800 (Pacific Standard Time) Embedded stack arn:aws:cloudformation:us-west-2:707129987630:stack/amplify-fox-dev-160955-apifox-Y0QXBP993EX1/bbfb9af0-263a-11ec-9178-0682b490ec17 was not successfully updated. Currently in UPDATE_ROLLBACK_IN_PROGRESS with reason: The following resource(s) failed to update: [SearchableStack].
```

**Comment by cjihrig** [comment]
@Rafcin there is a proposed fix in https://github.com/aws-amplify/amplify-cli/pull/9491. Until that is released, please downgrade to a version prior to v7.6.7.

**Comment by Rafcin** [comment]
I don't wanna be that guy, but 7.6.9 gave me so funky IAM/OS errors :(
```
UPDATE_FAILED               OpenSearchStreamingLambdaFunction6ED4A1FD                               AWS::Lambda::Function      Thu Jan 13 2022 16:56:13 GMT-0800 (Pacific Standard Time) OpenSearchStreamingLambdaFunction already exists in stack arn:aws:cloudformation:us-west-2:707129987630:stack/amplify-fox-prod-174207-apifox-1XCP8S1CGPDTM-SearchableStack-16OIOH0271T9U/63e749e0-263f-11ec-baa0-0a7614ed120d
UPDATE_FAILED               OpenSearchAccessIAMRole6A1D9CC5                                         AWS::IAM::Role             Thu Jan 13 2022 16:56:13 GMT-0800 (Pacific Standard Time) Resource creation cancelled                                                                                                                                                                                                  
UPDATE_ROLLBACK_IN_PROGRESS amplify-fox-dev-160955-apifox-Y0QXBP993EX1-SearchableStack-TV78E7975PIC AWS::CloudFormation::Stack Thu Jan 13 2022 16:56:14 GMT-0800 (Pacific Standard Time) The following resource(s) failed to update: [OpenSearchAccessIAMRole6A1D9CC5, OpenSearchStreamingLambdaFunction6ED4A1FD].                                                                                                    
⠼ Updating resources in the cloud. This may take a few minutes...

UPDATE_FAILED               SearchableStack                            AWS::CloudFormation::Stack Thu Jan 13 2022 16:56:19 GMT-0800 (Pacific Standard Time) Embedded stack arn:aws:cloudformation:us-west-2:707129987630:stack/amplify-fox-dev-160955-apifox-Y0QXBP993EX1-SearchableStack-TV78E7975PIC/ec47cd50-263a-11ec-bcd6-0a9127743e0d was not successfully updated. Currently in UPDATE_ROLLBACK_IN_PROGRESS with reason: The following resource(s) failed to update: [OpenSearchAccessIAMRole6A1D9CC5, OpenSearchStreamingLambdaFunction6ED4A1FD]. 
UPDATE_ROLLBACK_IN_PROGRESS amplify-fox-dev-160955-apifox-Y0QXBP993EX1 AWS::CloudFormation::Stack Thu Jan 13 2022 16:56:20 GMT-0800 (Pacific Standard Time) The following resource(s) failed to update: [SearchableStack].                                                                                                                                                                                                                                                                                                                                
⠙ Updating resources in the cloud. This may take a few minutes...

UPDATE_FAILED               apifox                 AWS::CloudFormation::Stack Thu Jan 13 2022 16:56:34 GMT-0800 (Pacific Standard Time) Embedded stack arn:aws:cloudformation:us-west-2:707129987630:stack/amplify-fox-dev-160955-apifox-Y0QXBP993EX1/bbfb9af0-263a-11ec-9178-0682b490ec17 was not successfully updated. Currently in UPDATE_ROLLBACK_IN_PROGRESS with reason: The following resource(s) failed to update: [SearchableStack]. 
```

**Comment by mattfysh** [comment]
@Rafcin I don't think the fix has been released yet, 7.6.9 was released 21 hours ago, but the fix was only merged 9 hours ago

**Comment by Rafcin** [comment]
@ngnathan How did you get this to work when downgrading to 7.6.5? I tried and it's not reverting the lambda name to the old way.

@lazpavel Can you reopen this until the CLI receives the update? Just to make sure this issue isn't forgotten. It impacts all users using OS.

**Comment by ngnathan** [comment]
@Rafcin I downgraded to 7.6.5, and then ran `amplify push --force` to ensure it runs the update on the `api`.  Then if you're using the Amplify Github Hosted flow, in the Amplify console >> Build settings, you'll need to make sure you set the Amplify CLI version to 7.6.5 as well, so that when it runs its own push, it doesn't replace the current lambda again.

**Comment by Rafcin** [comment]
Thank you! I'll test this out now 😄

**Comment by Rafcin** [comment]
@ngnathan I tried that, and it gives me more errors :(
```
UPDATE_FAILED      SearchableRestaurantLambdaMappingB7879B15 AWS::Lambda::EventSourceMapping Fri Jan 14 2022 23:39:30 GMT-0800 (Pacific Standard Time) Resource handler returned message: "User: arn:aws:sts::707129987630:assumed-role/us-west-2_Dpvj9kGVp_Full-access/amplifyadmin is not authorized to perform: lambda:UpdateEventSourceMapping on resource: arn:aws:lambda:us-west-2:707129987630:event-source-mapping:a03daf78-583f-404a-bb80-e5ed1d2730b0 because no identity-based policy allows the lambda:UpdateEventSourceMapping action (Service: Lambda, Status Code: 403, Request ID: a7c21066-c8cb-4557-b8a5-46817cfb8dbe, Extended Request ID: null)" (RequestToken: cb96b09d-d01c-ba16-2496-5886ea2ea189, HandlerErrorCode: GeneralServiceException)
UPDATE_FAILED      SearchableUserLambdaMapping231B0978       AWS::Lambda::EventSourceMapping Fri Jan 14 2022 23:39:31 GMT-0800 (Pacific Standard Time) Resource handler returned message: "User: arn:aws:sts::707129987630:assumed-role/us-west-2_Dpvj9kGVp_Full-access/amplifyadmin is not authorized to perform: lambda:UpdateEventSourceMapping on resource: arn:aws:lambda:us-west-2:707129987630:event-source-mapping:ad6b2f45-06b0-4107-949b-cfb526363536 because no identity-based policy allows the lambda:UpdateEventSourceMapping action (Service: Lambda, Status Code: 403, Request ID: aa29c9d7-5528-4c8c-8038-5d30a4d0ebad, Extended Request ID: null)" (RequestToken: f553564a-5261-4594-1860-4a2652af1927, HandlerErrorCode: GeneralServiceException)
⠹ Updating resources in the cloud. This may take a few minutes...

UPDATE_ROLLBACK_IN_PROGRESS amplify-fox-prod-174207-apifox-1XCP8S1CGPDTM-SearchableStack-16OIOH0271T9U AWS::CloudFormation::Stack Fri Jan 14 2022 23:39:32 GMT-0800 (Pacific Standard Time) The following resource(s) failed to update: [SearchableUserLambdaMapping231B0978, SearchableRestaurantLambdaMappingB7879B15]. 
⠏ Updating resources in the cloud. This may take a few minutes...

UPDATE_FAILED               SearchableStack                              AWS::CloudFormation::Stack Fri Jan 14 2022 23:39:45 GMT-0800 (Pacific Standard Time) Embedded stack arn:aws:cloudformation:us-west-2:707129987630:stack/amplify-fox-prod-174207-apifox-1XCP8S1CGPDTM-SearchableStack-16OIOH0271T9U/63e749e0-263f-11ec-baa0-0a7614ed120d was not successfully updated. Currently in UPDATE_ROLLBACK_IN_PROGRESS with reason: The following resource(s) failed to update: [SearchableUserLambdaMapping231B0978, SearchableRestaurantLambdaMappingB7879B15]. 
UPDATE_ROLLBACK_IN_PROGRESS amplify-fox-prod-174207-apifox-1XCP8S1CGPDTM AWS::CloudFormation::Stack Fri Jan 14 2022 23:39:46 GMT-0800 (Pacific Standard Time) The following resource(s) failed to update: [SearchableStack].                                                                                                                                                                                                                                                                                                                                       
⠴ Updating resources in the cloud. This may take a few minutes...

UPDATE_FAILED apifox AWS::CloudFormation::Stack Fri Jan 14 2022 23:40:01 GMT-0800 (Pacific Standard Time) Embedded stack arn:aws:cloudformation:us-west-2:707129987630:stack/amplify-fox-prod-174207-apifox-1XCP8S1CGPDTM/0ff51d80-263f-11ec-bda8-066ebf633925 was not successfully updated. Currently in UPDATE_ROLLBACK_IN_PROGRESS with reason: The following resource(s) failed to update: [SearchableStack]. 
```

**Comment by cjihrig** [comment]
It looks like you need permission to perform `lambda:UpdateEventSourceMapping`.

**Comment by Rafcin** [comment]
@cjihrig It's weird, the role I'm signed in with has full access, unless it just happens to omit that specific permission.

**Comment by mattfysh** [comment]
Please deploy the fix ASAP, downgrading to 7.6.5 introduces other issues and is not an acceptable workaround

**Comment by ShenShu2016** [comment]
update amplify Cli to beta version:

`sudo npm install -g @aws-amplify/cli@7.6.10`

and it works

But it need extra permission for AdministratorAccess-Amplify

"Effect": "Allow"
"Action": "lambda:UpdateEventSourceMapping"

At the end its worked well

**Comment by lazpavel** [comment]
Fix released in 7.6.10
